dozer/cogs/levels.py

```diff
@@ -0,0 +1,584 @@
+        await self.bot.wait_until_ready()
+
+    def cog_unload(self):
+        """Detach from the running bot and cancel long-running code as the cog is unloaded."""
+        self.sync_task.stop()
+
+    def _ensure_sync_running(self):
+        task = self.sync_task.get_task()
+        if task is None or not task.done():  # has not been started or has been started and not stopped
+            return
+        try:
+            exc = task.exception()
+        except asyncio.CancelledError:
+            logger.warning("Task syncing records was cancelled prematurely, restarting")
+        else:
+            # exc could be None if the task returns normally, but that would also be an error
+            logger.error("Task syncing records failed: %r", exc)
+        finally:
+            self.sync_task.start()
+
+    @command()
+    @guild_only()
+    @has_permissions(administrator=True)
+    async def setrolelevel(self, ctx, role: discord.Role, level: int):
+        """Sets a role to be given to a user when they reach a certain level"""
+        if role > ctx.author.top_role:
+            raise BadArgument('Cannot give roles higher than your top role!')
+
+        if role > ctx.me.top_role:
+            raise BadArgument('Cannot give roles higher than my top role!')
+
+        if level <= 0:
+            raise BadArgument("Cannot give level roles lower and/or equal to zero!")
+
+        if role == ctx.guild.default_role:
+            raise BadArgument("Cannot give @\N{ZERO WIDTH SPACE}everyone for a level")
+
+        async with ctx.channel.typing():  # Send typing to show that the bot is thinking and not stalled
+            ent = XPRole(
+                guild_id=int(ctx.guild.id),
+                role_id=int(role.id),
+                level=int(level)
+            )
+
+            await ent.update_or_add()
+
+            await self.update_level_role_cache()
+
+            e = discord.Embed(color=blurple)
+            e.add_field(name='Success!', value=f"{role.mention} will be given to users who reach level {level}")
+            e.set_footer(text='Triggered by ' + ctx.author.display_name)
+            await ctx.send(embed=e)
+
+    setrolelevel.example_usage = """
+            `{prefix}setrolelevel "level 2" 2`: Will configure the role level 2 to be given to users who reach level 2` 
+            """
+
+    @command(aliases=["configurelevels"])
+    @guild_only()
+    @has_permissions(administrator=True)
+    async def configureranks(self, ctx):
+        """Configure dozer ranks:tm: any key word arguments not entered will be treated as default"""
+        args = self.kwarg_parsing(ctx.message.content)  # Parse for kwargs
+
+        xp_min = max(min(int(args.get("min")), 32767), 0) if args.get("min") else 5
+        xp_max = max(min(int(args.get("max")), 32767), 0) if args.get("max") else 15
+        xp_cooldown = max(min(int(args.get("cooldown")), 32767), 0) if args.get("cooldown") else 15
```

Comment:
Clamping the value into range is likely not what the user expected or wanted. If these values are out of bounds, the command should error.